File: Ex01/main.py
from threading import Thread

i = 0

def someThreadFunction():
    global i

    print("Hello from thread 1!")

    for j in range(1000000):
        ++i

def someThreadFunction2():
    global i

    print("Hello from thread 2!")

    for j in range(1000000):
        --i


def main():
    someThread = Thread(target = someThreadFunction, args = (),)
    someThread2 = Thread(target = someThreadFunction2, args = (),)

    someThread.start()
    someThread2.start()

    someThread.join()
    someThread2.join()

    print("Hello from main!")
    print(i)


main()

File: Ex01/main.c
#include <pthread.h>
#include <stdio.h>

int i = 0;

void* someThreadFunction(){

    int j;

    for (j = 0; j < 1000000; ++j)
        ++i;

    printf("Hello from thread 1!\n");
    return NULL;
}

void* someThreadFunction2(){

    int j;

    for (j = 0; j < 1000000; ++j)
        --i;

    printf("Hello from thread2!\n");
    return NULL;
}



int main()
{

    pthread_t someThread;
    pthread_t someThread2;

    pthread_create(&someThread, NULL, someThreadFunction, NULL);
    pthread_create(&someThread2, NULL, someThreadFunction2, NULL);

    pthread_join(someThread, NULL);
    pthread_join(someThread2, NULL);

    printf("i = %d\n", i);

    return 0;
}
